### ISSUE TITLE
Accessibility of the UI

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5-table#31]
ATM
* an error is thrown when navigating to the insertion panel https://github.com/ckeditor/ckeditor5-table/issues/30,
* there's no strategy as to how the a11y should work in the panel; probably the arrow keys will control the geometry of the grid but we also need to understand how to announce that,
* there's no way to focus the table toolbar using the keyboard and use its features (a twin of ckeditor5-image [issue](https://github.com/ckeditor/ckeditor5-image/issues/203), we need a general strategy)

**Comment by jodator** [comment]
* The insertion panel requires defining how the keyboard support should be implemented: #22.

**Comment by juleskuehn** [comment]
I'd like to see the table button open a dialog in which you can specify "rows" and "columns" in text inputs.
The other issues can be resolved by including the pop-up table toolbar buttons in the main toolbar, correct?

**Comment by oleq** [comment]
What do you mean by

> pop-up table toolbar buttons 

@juleskuehn ?

**Comment by juleskuehn** [comment]
![Screenshot of pop-up table toolbar buttons](https://user-images.githubusercontent.com/1150048/61725732-53fa4900-ad3e-11e9-9a4e-57e3eced071f.png)
@oleq Floating menu that appears when a table is focused

**Comment by oleq** [comment]
> I'd like to see the table button open a dialog in which you can specify "rows" and "columns" in text inputs.

Check out https://github.com/ckeditor/ckeditor5-table/issues/22.

> The other issues can be resolved by including the pop-up table toolbar buttons in the main toolbar, correct?

Not sure what you mean here. You'd expect us to drop the balloon completely or move its content to the main toolbar in certain situations? If the later, what would it look/work like?

**Comment by juleskuehn** [comment]
The buttons in the pop-up toolbar are not keyboard accessible, but those buttons can already be included in the main toolbar, where they are accessible. This is a workaround I'm using to make these features accessible (although I think a keyboard-only user would still be frustrated by the existence of the inaccessible pop-up menu). I don't have a solution for the pop-up menu other than adding a keyboard shortcut to focus pop-up menus. I'm just bringing it up for discussion.

Two remaining issues.
1. Keyboard focusing the main toolbar (via Alt+F10) does not dismiss the pop-up toolbar, which obscures the dropdown from the row / column buttons:
![image](https://user-images.githubusercontent.com/1150048/61797760-0ccf8f00-adf6-11e9-9b6d-322b69f483db.png)
2. When "Header row" or "Header column" are toggled, focus moves back to the content editing area, but the drop-down menu containing the toggle switch remains onscreen.
![image](https://user-images.githubusercontent.com/1150048/61798025-92ebd580-adf6-11e9-9047-f7a89a13ad1e.png)

**Comment by oleq** [comment]
Moving these buttons to the main toolbar seems to be an overkill. The real issue is that there's no way to focus widget toolbar and this should be addressed.

> there's no way to focus the table toolbar using the keyboard and use its features (a twin of ckeditor5-image issue, we need a general strategy)

@mlewand Do you think we can work on this anytime soon?

**Comment by mlewand** [comment]
I agree with @oleq here. There's a reason why we placed these buttons in a separate toolbar.

Our goal is to avoid cluttering the UI (in this case main toolbar) with extra buttons if possible. That's why we have these context-based additional balloon toolbars.

This particular issue should be simply fixed by providing a way to move the focus into the balloon toolbar (which we need anyway, because it will also apply to [the image toolbar](https://github.com/ckeditor/ckeditor5-image/issues/203)). My first idea is to make our `alt` + `f10` to cycle across all available toolbars (and `alt` + `shift` + `f10` in reverse direction).

As for expected ETA I don't have any at this moment. But please, add a 👍 reaction to the main post of this issue so that we can see that community is interested in this improvement.

**Comment by oleq** [comment]
> This particular issue should be simply fixed by providing a way to move the focus into the balloon toolbar (which we need anyway, because it will also apply to [the image toolbar](https://github.com/ckeditor/ckeditor5-image/issues/203)). My first idea is to make our `alt` + `f10` to cycle across all available toolbars (and `alt` + `shift` + `f10` in reverse direction).

cc @Comandeer
